src/components/transactions/TxDetails/TxData/ExecTransaction/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { Safe__factory } from '@/types/contracts'
+import { Skeleton, Stack, SvgIcon, Typography } from '@mui/material'
+import { getConfirmationView, type TransactionData } from '@safe-global/safe-gateway-typescript-sdk'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import { MethodDetails } from '../DecodedData/MethodDetails'
+import MethodCall from '../DecodedData/MethodCall'
+import DecodedTx, { Divider } from '@/components/tx/DecodedTx'
+import Link from 'next/link'
+import { useCurrentChain } from '@/hooks/useChains'
+import { AppRoutes } from '@/config/routes'
+import { useMemo } from 'react'
+import type { SafeTransaction } from '@safe-global/safe-core-sdk-types'
+import useAsync from '@/hooks/useAsync'
+import NestedTransactionIcon from '@/public/images/transactions/nestedTx.svg'
+import ExternalLink from '@/components/common/ExternalLink'
+
+const safeInterface = Safe__factory.createInterface()
+
+const extractTransactionData = (data: string | undefined): SafeTransaction | undefined => {
+  const params = data ? safeInterface.decodeFunctionData('execTransaction', data) : undefined
+  if (!params || params.length !== 10) {
+    return
+  }
+
+  return {
+    addSignature: () => {},
+    encodedSignatures: () => params[9],
+    getSignature: () => undefined,
+    data: {
+      to: params[0],
+      value: params[1],
+      data: params[2],
+      operation: params[3] as number,
+      safeTxGas: params[4],
+      baseGas: params[5],
+      gasPrice: params[6],
+      gasToken: params[7],
+      refundReceiver: params[8],
+      nonce: 0,
+    },
+    signatures: new Map(),
+  }
+}
+
+export const ExecTransaction = ({ data }: { data?: TransactionData }) => {
+  const chain = useCurrentChain()
+
+  const childSafeTx = useMemo<SafeTransaction | undefined>(() => extractTransactionData(data?.hexData), [data?.hexData])
+
+  const [dataDecoded, error] = useAsync(async () => {
+    if (chain?.chainId && data?.to.value && childSafeTx) {
+      return await getConfirmationView(
```

Comment:
Can we try getting this from the preview endpoint instead? We want to eventually [deprecate the confirmation view](https://github.com/safe-global/safe-wallet-web/issues/4331).

Reply:
I tried to change it quickly but the returned types are not fully compatible. So we need to adjust the component that renders the result.

Lets do it in a separate tech debt issue to reduce the scope.